### PR TITLE
feat(dashboards): add overlay to create dashboard from template

### LINKED ISSUE #12913: Add import from template option to create resource dropdown
Add `Import from template` option to dashboard create button. 

Selecting this option opens an overlay that shows cards for available templates of that resource type:
https://xd.adobe.com/view/4e62c4ff-ef59-4e4d-7061-079197eb3130-3853/?fullscreen

in this issue, create the dropdown option
clicking the dropdown option routes to `.../templates`
route renders overlay.

### PR DESCRIPTION
**Pull request by AlirieGray**
Closes #12913

This PR adds an overlay to create a new dashboard from a template.

  - [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/dashboards/components/dashboard_index/Table.tsx [hunk 1]
```diff
@@ -124,6 +124,14 @@ class DashboardsTable extends PureComponent<Props, State> {
     this.setState({sortKey, sortDirection: nextSort, sortType})
   }
 
+  private summonImportFromTemplateOverlay = (): void => {
+    const {
+      router,
+      params: {orgID},
+    } = this.props
+    router.push(`/orgs/${orgID}/dashboards/templates`)
```
Comment (bthesorceror) [inline ui/src/dashboards/components/dashboard_index/Table.tsx]: this route url should have something in it to indicate that its import `/orgs/${orgID}/dashboards/import/template` would be my go to.

File: ui/src/dashboards/components/DashboardImportFromTemplateOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+                  </Panel.Body>
+                </Panel>
+              )}
+              {selectedTemplateSummary && (
+                <Panel className="import-template-overlay--details">
+                  <Panel.Header
+                    title={_.get(selectedTemplateSummary, 'meta.name')}
+                  />
+                  <Panel.Body>
+                    {/* TODO: Add Template description */}
```
Comment (bthesorceror) [inline ui/src/dashboards/components/DashboardImportFromTemplateOverlay.tsx]: Please do not leave comments in JSX

File: ui/src/dashboards/actions/index.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 // Libraries
 import {Dispatch} from 'redux'
+import _ from 'lodash'
```
Comment (bthesorceror) [inline ui/src/dashboards/actions/index.ts]: Do we need lodash here? Nothing else added seems to use it.

File: ui/src/dashboards/components/DashboardImportFromTemplateOverlay.tsx [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+          />
+          <Overlay.Body>
+            <div className="import-template-overlay">
+              <OrgTemplateFetcher orgName={this.orgName}>
+                <ResponsiveGridSizer columns={3}>
+                  {this.templates}
+                </ResponsiveGridSizer>
+              </OrgTemplateFetcher>
+              {!selectedTemplateSummary && (
+                <Panel className="import-template-overlay--empty">
```
Comment (bthesorceror) [inline ui/src/dashboards/components/DashboardImportFromTemplateOverlay.tsx]: Could we extract the empty state? There is a lot going on in this render function.

File: ui/src/organizations/components/Dashboards.tsx [hunk 5]
```diff
@@ -117,6 +118,11 @@ class Dashboards extends PureComponent<Props, State> {
     router.push(`/organizations/${params.orgID}/dashboards/import`)
   }
 
+  private summonImportFromTemplateOverlay = (): void => {
+    const {router, params} = this.props
+    router.push(`/organizations/${params.orgID}/dashboards/templates`)
```
Comment (bthesorceror) [inline ui/src/organizations/components/Dashboards.tsx]: Same comment about the route.

File: ui/src/shared/components/AddResourceDropdown.tsx [hunk 6]
```diff
@@ -34,14 +35,30 @@ export default class AddResourceDropdown extends PureComponent<Props> {
...
+    const items = [
       <Dropdown.Item id={newOption} key={newOption} value={newOption}>
         {newOption}
       </Dropdown.Item>,
       <Dropdown.Item id={importOption} key={importOption} value={importOption}>
         {importOption}
       </Dropdown.Item>,
     ]
+
+    if (!!this.props.onSelectTemplate) {
```
Comment (bthesorceror) [inline ui/src/shared/components/AddResourceDropdown.tsx]: seems odd that the existence of a handler is determining whether to render this.

File: ui/src/organizations/components/Dashboards.tsx (hunk 5 above)
Comment (AlirieGray) [reply ui/src/organizations/components/Dashboards.tsx]: would you recommend a boolean prop instead?

File: ui/src/shared/components/AddResourceDropdown.tsx (hunk 6 above)
Comment (AlirieGray) [reply ui/src/shared/components/AddResourceDropdown.tsx]: would you recommend a boolean prop instead?

Comment (bthesorceror) [reply ui/src/shared/components/AddResourceDropdown.tsx]: yeah that seems more informative

**Pull request merged**